integrations/helm/chartsync/chartsync.go

```diff
@@ -202,6 +204,11 @@ func (chs *ChartChangeSync) Run(stopCh <-chan struct{}, errc chan error, wg *syn
 						chs.logger.Log("warning", "could not get revision for ref while checking for changes", "repo", repoURL, "ref", ref, "err", err)
 						continue
 					}
+					err = chs.statusUpdater.UpdateReleaseRevision(fhr, refHead)
```

Comment:
What's the rationale for updating the status here? We haven't even looked at what revision might be released, at this point. If the intention of the ReleaseRevision is to keep track of the most recently released chart revision (/version), then `reconcileReleaseDef` (later in the file, around L350) is probably a better place to do it, since

 - that's where the release is attempted
 - all the chart releasing is funneled through there -- i.e., any change to which version of the chart is released will go through there.

(But I may be wrong about the intention!)

Reply:
You're right :)